From c02486ebfdf3a1448fdd538c16d2a28f6f1d4110 Mon Sep 17 00:00:00 2001
From: Sen Jiang <senj@google.com>
Date: Wed, 2 Jan 2019 15:54:40 -0800
Subject: [PATCH 3/3] Android: Reset update progress when booted into new
 version.

In Android we don't reset progress after update completes because if
automatic update is toggled off after that, we will need to revert the
slot switch and still keep the progress.

However if we try to apply the same full update again after booted
into the new version, it will try to resume and fail on verification
because the inactive slot is now the previous version, so we shouldn't
resume in this case.

Bug: 122117961
Test: progress is reset after booted into new version
Change-Id: I394a78fd37bbbdcf62bf85639a9f86e7d3bc42c6
---
 update_attempter_android.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/update_attempter_android.cc b/update_attempter_android.cc
index 42fd63d..bb2d86a 100644
--- a/update_attempter_android.cc
+++ b/update_attempter_android.cc
@@ -827,6 +827,11 @@ void UpdateAttempterAndroid::UpdatePrefsAndReportUpdateMetricsOnReboot() {
   metrics_utils::LoadAndReportTimeToReboot(
       metrics_reporter_.get(), prefs_, clock_.get());
   ClearMetricsPrefs();
+
+  // Also reset the update progress if the build version has changed.
+  if (!DeltaPerformer::ResetUpdateProgress(prefs_, false)) {
+    LOG(WARNING) << "Unable to reset the update progress.";
+  }
 }
 
 // Save the update start time. Reset the reboot count and attempt number if the
-- 
2.17.1

